lib/redmine_email_inline_images/mail_handler_patch.rb

```diff
@@ -30,14 +30,21 @@ def plain_text_body_with_email_inline_images
         # replace html images with text bang notation
         email_doc = Nokogiri::HTML(@plain_text_body)
         email_doc.xpath('//img').each do |image|
-            image_bang = "\n!#{email_images[image['src']]}!"
-            image.replace(email_doc.create_text_node(image_bang))
+            case Setting.text_formatting
+            when 'markdown'
+                image_bang = "\n![](#{email_images[image['src']]})"
+            when 'textile'
+                image_bang = "\n!#{email_images[image['src']]}!"
+            else
+                image_bang = nil
+            end
+            image.replace(email_doc.create_text_node(image_bang)) if image_bang
         end
         @plain_text_body = email_doc.to_html
         
         # strip html tags and remove doctype directive
-        @plain_text_body = strip_tags(@plain_text_body.strip)
-        @plain_text_body.sub! %r{^<!DOCTYPE .*$}, ''
+        @plain_text_body = self.class.html_body_to_text(@plain_text_body)
```

Comment:
There is a native Redmine method for stripping HTML tags - MailHandler's #html_body_to_text. I believe, this method should be used here.